File: msafe/msafe.cpp
#include <stdio.h>

int n, k, cleft;
int a[10000001];
long long sleft, sright, res;

void input() {
    scanf("%d\n", &k);

    n = 10000000;
    int num;
    for (int i = 1; i <= k ; i++) {
        scanf("%d\n", &num);    
        
        if (num == 1) {
            cleft++;
            a[1]++;
            continue;
        }
        
        if (num > 5000000) {
            cleft++;
            sleft += (n + 1 - num);
        }
        else 
            sright += (num - 1);
            
        a[num]++;
    }
}

void process() {
    res = sleft + sright;
    
    int j;
    for (int i = 2; i <= n; i++) {
        j = i + 4999999;
        if (j > n) j -= n;
    
        sleft += cleft - a[j] * 5000001;
        sright += a[j] * 4999999 - (k - cleft);
        cleft += a[i] - a[j];
            
        if (sleft + sright < res)
            res = sleft + sright;
    }
    
   printf("%lld\n", res);
}

int main() {
//    freopen("msafe.in1", "r", stdin);
//    freopen("msafe.out", "w", stdout);
    
    input();
    process();    
    
    return 0;
}

File: msafe.pas
uses math;
const
	inputfile = '';
  outputfile = '';
  maxN = 10000000;
//type

var
	fi,fo: text;
  res: int64;
  n,k: longint;

  a: array[1..maxN] of longint;

  cleft,cright: longint;
  sleft,sright: int64;


procedure openfile;
begin
	assign(fi,inputfile); reset(fi);
  assign(fo,outputfile); rewrite(fo);
end;

procedure closefile;
begin
	close(fi);
  close(fo);
end;

procedure input;
var
	i,num: longint;
begin
  n:= 10000000;

  //
	readln(fi,k);
  for i:= 1 to k do
  begin
    readln(fi,num);
    if (num = 1) then
    begin
    	inc(cleft);
    	inc(a[1]);
      continue;
    end;
		
    if (num > 5000000) then
    begin
    	inc(cleft);
    	sleft:= sleft + (n + 1 - num);
    end
    else
    	sright:= sright + (num - 1);

    inc(a[num]);
  end;
end;

procedure process;
var
	i,j: longint;
begin
  res:= sleft + sright;

  //
  for i:= 2 to n do
  begin
  	j:= i + 4999999;
	  if (j > n) then j:= j - n;

    //
  	sleft:= sleft - a[j] * 5000001 + cleft;
  	sright:= sright - (k - cleft) + a[j] * 4999999;
    cleft:= cleft - a[j] + a[i];

    if (sleft + sright < res) then
    	res:= sleft + sright;
  end;

  writeln(fo,res);
end;

procedure output;
begin
end;


BEGIN
	openfile;
  input;
  process;
  output;
  closefile;
END.
